### PR TITLE
[`fix:`]: initialize `ModernBertUnpaddedRotaryEmbedding` when loading from a checkpoint

### PR DESCRIPTION
**Pull request by emapco**
# What does this PR do?

 This fix improves accuracy and prevents existing ModernBert models from yielding NaN tensors. Previously, when a pretrained  modernbert model is loaded, the `ModernBertUnpaddedRotaryEmbedding`'s `inv_freq` non-persistent buffer is not initialized resulting in `NaN` loss and logits.

Without fix: 
- `ModernBertForSequenceClassification`  `ModernBertModelIntegrationTest.test_inference_sequence_classification_flash_attention_2` rarely returns NaN tensors but instead fails due to inaccurate logits
- `ModernBertModelIntegrationTest.test_inference_sequence_classification_flash_attention_2_modernbert_base` always fails due to NaN logits.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [x] Did you write any new necessary tests?

## Who can review?

@ArthurZucker @Cyrilvallez

### CODE REVIEW

**Comment by emapco** [comment]
I should add that I used the `flash-attn` package that I installed via `uv pip install flash-attn --no-build-isolation` with `torch==2.9` and `cuda12.8`.

**Comment by Cyrilvallez** [comment]
Fixed in https://github.com/huggingface/transformers/pull/43115!
